File: app/src/main/java/com/young/simpledict/netsoundplayer/request/NetSoundRequest.java
package com.young.simpledict.netsoundplayer.request;

import com.young.simpledict.service.event.BaseEvent;

/**
 * Author: taylorcyang
 * Date:   2014-10-29
 * Time:   12:11
 * Life with passion. Code with creativity!
 */
public class NetSoundRequest extends BaseEvent {
    public static final int REQUEST_START = 1;
    public static final int REQUEST_PAUSE = 2;
    public static final int REQUEST_STOP = 3;
    public static final int REQUEST_RESUME = 4;

    public String soundUrl;
    public int requestCode = REQUEST_START;
    public boolean releaseAllMediaPlayer = false;
}

File: app/src/main/java/com/young/simpledict/netsoundplayer/request/NetSoundRequest.kt
package com.young.simpledict.netsoundplayer.request

import com.young.simpledict.service.event.BaseEvent

/**
 * Author: landerlyang
 * Date:   2014-10-29
 * Time:   12:11
 * Life with passion. Code with creativity!
 */
class NetSoundRequest(
    eventCode: Any, var soundUrl: String,
    var requestCode: Int,
    var releaseAllMediaPlayer: Boolean
) : BaseEvent(eventCode) {

    constructor(eventCode: Any, soundUrl: String) : this(eventCode, soundUrl, REQUEST_START, false)

    companion object {
        const val REQUEST_START = 1
        const val REQUEST_PAUSE = 2
        const val REQUEST_STOP = 3
        const val REQUEST_RESUME = 4
    }
}
